### PR TITLE
Merge 3.0 into main

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: src/Metadata/HttpOperation.php [hunk 1]
```diff
@@ -95,15 +96,13 @@ public function __construct(
         protected ?string $condition = null,
         protected ?string $controller = null,
         protected ?array $cacheHeaders = null,
-
+        protected ?array $paginationViaCursor = null,
         protected ?array $hydraContext = null,
-        protected ?array $openapiContext = null, // TODO Remove in 4.0
```
Comment (vincentchalamon) [inline src/Metadata/HttpOperation.php:100]: Rollback this removal of todo

File: src/Metadata/HttpOperation.php [hunk 2]
```diff
@@ -529,12 +513,12 @@ public function withOpenapiContext(array $openapiContext): self
         return $self;
     }
 
-    public function getOpenapi(): bool|OpenApiOperation|null
```
Comment (vincentchalamon) [inline src/Metadata/HttpOperation.php:532]: Rollback

File: src/Metadata/Put.php [hunk 3]
```diff
@@ -100,9 +94,80 @@ public function __construct(
         array $extraProperties = [],
         private ?bool $allowCreate = null,
     ) {
-        $args = \func_get_args();
-        parent::__construct(self::METHOD_PUT, ...$args);
-        $this->allowCreate = $allowCreate;
```
Comment (vincentchalamon) [inline src/Metadata/Put.php:105]: Rollback or edit to keep this property

File: src/Metadata/Resource/Factory/OperationDefaultsTrait.php [hunk 4]
```diff
@@ -208,7 +208,7 @@ private function getDefaultOperationName(HttpOperation $operation, string $resou
         return sprintf(
             '_api_%s_%s%s',
             $path ?: ($operation->getShortName() ?? $this->getDefaultShortname($resourceClass)),
-            strtolower($operation->getMethod() ?? HttpOperation::METHOD_GET),
```
Comment (vincentchalamon) [inline src/Metadata/Resource/Factory/OperationDefaultsTrait.php:211]: Not sure about this change, should it be rollbacked too?

File: src/OpenApi/Factory/OpenApiFactory.php [hunk 5]
```diff
@@ -477,7 +477,7 @@ private function getLinks(ResourceMetadataCollection $resourceMetadataCollection
                     $operationName === $operation->getName() ||
                     isset($links[$operationName]) ||
                     $operation instanceof CollectionOperationInterface ||
-                    HttpOperation::METHOD_GET !== ($method ?? null)
```
Comment (vincentchalamon) [inline src/OpenApi/Factory/OpenApiFactory.php:480]: Same: not sure about this change, should it be rollbacked too?

File: src/Symfony/Bundle/Resources/config/api.xml [hunk 6]
```diff
@@ -168,8 +168,10 @@
         </service>
         <service id="ApiPlatform\Api\IriConverterInterface" alias="api_platform.symfony.iri_converter" />
 
-        <service id="api_platform.error_listener" class="ApiPlatform\Symfony\EventListener\ErrorListener" parent="exception_listener">
```
Comment (vincentchalamon) [inline src/Symfony/Bundle/Resources/config/api.xml:171]: The ErrorListener still extends the Symfony one: https://github.com/api-platform/core/blob/main/src/Symfony/EventListener/ErrorListener.php

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/Resources/config/api.xml (hunk 6 above)
Comment (soyuka) [reply src/Symfony/Bundle/Resources/config/api.xml:171]: this was changed on 2.7

File: src/Metadata/Put.php (hunk 3 above)
Comment (soyuka) [reply src/Metadata/Put.php:105]: me moved it to the constructor

File: src/Metadata/Resource/Factory/OperationDefaultsTrait.php (hunk 4 above)
Comment (soyuka) [reply src/Metadata/Resource/Factory/OperationDefaultsTrait.php:211]: its fine as well an http operation has always a method

File: src/OpenApi/Factory/OpenApiFactory.php (hunk 5 above)
Comment (soyuka) [reply src/OpenApi/Factory/OpenApiFactory.php:480]: same an httpoperation always has a method

**Review by vincentchalamon — APPROVED** [review]
LGTM

**Pull request merged**